pandas/tests/series/test_missing.py

```diff
@@ -578,6 +578,28 @@ def test_fillna_categorical(self, fill_value, expected_output):
         exp = Series(Categorical(expected_output, categories=["a", "b"]))
         tm.assert_series_equal(s.fillna(fill_value), exp)
 
+    @pytest.mark.parametrize(
+        "new_categories, fill_value, expected_output",
+        [
+            (
+                ["c", "d", "e"],
+                Series(["a", "b", "c", "d", "e"]),
+                ["a", "b", "b", "d", "e"],
+            )
+        ],
+    )
+    def test_fillna_categorical_with_new_categories(
+        self, new_categories, fill_value, expected_output
```

Comment:
Why do you need `new_categories` at all? Can you not create the series with `categories=['a', 'b', 'c', 'd', 'e']`?

Can you add a test where the original series and the fill_value series have the same categories, but in different order?

```python
s = pd.Series(pd.Categorical(['a', None, 'b'], categories=['b', 'c', 'a']))
s.fillna(pd.Series(pd.Categorical(['a', 'c', 'a'], categories=['c', 'a', 'b']))
```